### PR TITLE
Itt 573/add banking webhook models

### PR DESCRIPTION
**Pull request by wboereboom**
## Summary
Adds all banking webhooks collections to the library, including handler functions per webhook type.

## Tested scenarios
Unit tests per webhook collection

### CODE REVIEW

**Review by michaelpaul — CHANGES_REQUESTED** [review]

File: src/adyen/api.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ const (
 type APIClient struct {
 	client *common.Client
 	// API Services
+
```
Comment (wboereboom) [inline src/adyen/api.go:82]:
```suggestion
```

**Review by jillingk — APPROVED** [review]

File: src/configurationwebhook/webhook_handler.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	res := PaymentNotificationRequest{}
+	err := json.Unmarshal([]byte(req), &res)
+	if err != nil {
+		return nil, err
+	}
+	return &res, nil
+}
+
+// HandleSweepconfigurationwebhookRequest creates a Notification object from the given JSON string
+func HandleSweepconfigurationwebhookRequest(req string) (*SweepconfigurationwebhookRequest, error) {
```
Comment (michaelpaul) [inline src/configurationwebhook/webhook_handler.go:48]: Something weird going on with this model's name. Looks like a find and replace took place here. Model names should be the same as in specs.

**Review by michaelpaul — CHANGES_REQUESTED** [review]

File: src/transferwebhook/client.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+// NewAPIClient creates a new API client.
+func NewAPIClient(client *common.Client) *APIClient {
+	c := &APIClient{}
+    c.common.Client = client
+    c.common.BasePath = func() string {
+        return client.Cfg.Endpoint
+    }
+
+	// API Services
```
Comment (michaelpaul) [inline src/transferwebhook/client.go:31]: This file is not necessary.

File: src/transferwebhook/webhook_handler.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package transferwebhook
+
+import (
+	"encoding/json"
+)
+
+// HandleTransferwebhookRequest creates a Notification object from the given JSON string
+func HandleTransferwebhookRequest(req string) (*TransferNotificationRequest, error) {
```
Comment (michaelpaul) [inline src/transferwebhook/webhook_handler.go:8]:
```suggestion
// HandleTransferNotificationRequest creates a Notification object from the given JSON string
func HandleTransferNotificationRequest(req string) (*TransferNotificationRequest, error) {
```

**Review by michaelpaul — CHANGES_REQUESTED** [review]
Please remove the `docs` folders. I've added a new badge in the readme which will take users to documentation pages more accurate (after tagging the release)  on https://pkg.go.dev/github.com/adyen/adyen-go-api-library/v7.

File: src/platformsnotificationevents/model_refund_funds_transfer_notification.go [hunk 5]
```diff
@@ -16,9 +16,9 @@ import (
 	"time"
 )
 
-// RefundFundsTransferNotification struct for RefundFundsTransferNotification
-type RefundFundsTransferNotification struct {
-	Content RefundFundsTransferNotificationContent `json:"content"`
+// RefundFundstransferwebhook struct for RefundFundstransferwebhook
+type RefundFundstransferwebhook struct {
```
[the remark this answers is not in the document]
Comment (wboereboom) [reply src/platformsnotificationevents/model_refund_funds_transfer_notification.go]: reverted these changes

**Review by michaelpaul — APPROVED** [review]

**Pull request merged**